xrootd/cmd/xrd-srv/main_test.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2018 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main // import "go-hep.org/x/hep/xrootd/cmd/xrd-srv"
+
+import (
+	"context"
+	"io/ioutil"
+	"net"
+	"os"
+	"path"
+	"reflect"
+	"sync"
+	"testing"
+
+	"github.com/pkg/errors"
+	"go-hep.org/x/hep/xrootd/client"
+	"go-hep.org/x/hep/xrootd/server"
+	"go-hep.org/x/hep/xrootd/xrdfs"
+	"go-hep.org/x/hep/xrootd/xrdproto"
+)
+
+const addr = "127.0.0.1:1094"
```

Comment:
we should probably randomize the port (otherwise, if that test is run concurrently by 2 instances, it will fail, as it did, actually :P)

I usually resort to this little function, that automatically returns a free port to use:
```go
func getTCPPort() (string, error) {
	addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
	if err != nil {
		return "", err
	}
	l, err := net.ListenTCP("tcp", addr)
	if err != nil {
		return "", err
	}
	defer l.Close()
	return strconv.Itoa(l.Addr().(*net.TCPAddr).Port), nil
}
```